src/Control/Retry.hs

```diff
@@ -221,15 +220,7 @@ toRetryAction False = DontRetry
 toRetryAction True = ConsultPolicy
 
 -------------------------------------------------------------------------------
--- | Datatype with stats about retries made thus far. The constructor
```

Comment:
The main point of this PR is to try to update this confusing out-of-date comment.
Since RetryStatus constructor has been exposed in https://github.com/Soostone/retry/commit/fc742cd69941f73957908333357c12f275398ce3#diff-8bf0263d4ec3b059959f2a1b0870986e2a3e8981a6dc495fe8091bb5d5900363R35 it adds more confusion than it exlains.